applications/RomApplication/costum_strategies/rom_builder_and_solver.h

```diff
@@ -0,0 +1,811 @@
+    typedef typename BaseType::NodesArrayType NodesArrayType;
+    typedef typename BaseType::ElementsArrayType ElementsArrayType;
+    typedef typename BaseType::ConditionsArrayType ConditionsArrayType;
+
+    /// Additional definitions
+    typedef PointerVectorSet<Element, IndexedObject> ElementsContainerType;
+    typedef Element::EquationIdVectorType EquationIdVectorType;
+    typedef Element::DofsVectorType DofsVectorType;
+    typedef boost::numeric::ublas::compressed_matrix<double> CompressedMatrixType;
+
+    /// DoF types definition
+    typedef Node<3> NodeType;
+    typedef typename NodeType::DofType DofType;
+    typedef typename DofType::Pointer DofPointerType;
+
+
+    /*@} */
+    /**@name Life Cycle
+     */
+    /*@{ */
+
+    /**
+     * @brief Default constructor. (with parameters)
+     */
+    explicit ROMBuilderAndSolver(typename TLinearSolver::Pointer pNewLinearSystemSolver, Parameters ThisParameters)
+    : BuilderAndSolver<TSparseSpace,TDenseSpace,TLinearSolver>(pNewLinearSystemSolver)
+    {
+        // Validate default parameters
+        Parameters default_parameters = Parameters(R"(
+        {
+            "nodal_unknowns" : [],
+            "number_of_rom_dofs" : 10
+        })" );
+
+        ThisParameters.ValidateAndAssignDefaults(default_parameters);
+
+        // We set the other member variables
+        mpLinearSystemSolver = pNewLinearSystemSolver;
+
+        mNodalVariablesNames = ThisParameters["nodal_unknowns"].GetStringArray();
+
+        mNodalDofs = mNodalVariablesNames.size();
+        mRomDofs = ThisParameters["number_of_rom_dofs"].GetInt();
+    }
+
+
+
+    /** Destructor.
+     */
+    virtual ~ROMBuilderAndSolver()
+    {
+    }
+
+
+    /*@} */
+    /**@name Operators
+     */
+	/*
+    virtual void SetUpDofSet(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& r_model_part
+    ) override
+    {
+        mDofList.clear();
+        for(auto& node : r_model_part.Nodes())
+        {
+            for(const std::string& var_name : mNodalVariablesNames)
+            {
+                if( KratosComponents< Variable<double> >::Has( var_name ) ) //case of double variable
+                {
+                    auto pdof = node.pGetDof( KratosComponents< Variable<double> >::Get( var_name ) );
+                    mDofList.push_back(pdof);
+                }
+                else if( KratosComponents< VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > >::Has(var_name) ) //case of component variable
+                {
+                    typedef VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > component_type;
+                    component_type var_component = KratosComponents< component_type >::Get(var_name);
+                    auto pdof = node.pGetDof( var_component.GetSourceVariable() );
+                    mDofList.push_back( pdof );
+                }
+            }
+        }
+    }
+	*/
+
+	virtual void SetUpDofSet(
```

Comment:
Please, pay attention to the general file formatting (4 spaces instead of tabs).